service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/digitaltwin/helpers/Tools.java

```diff
@@ -55,7 +56,7 @@
     public static final Func1<ServiceResponseWithHeaders<Object, DigitalTwinInvokeComponentCommandHeaders>, Observable<ServiceResponseWithHeaders<DigitalTwinCommandResponse, DigitalTwinInvokeCommandHeaders>>> FUNC_TO_DIGITAL_TWIN_COMPONENT_COMMAND_RESPONSE = object -> {
         try {
             DigitalTwinCommandResponse digitalTwinCommandResponse = new DigitalTwinCommandResponse();
-            digitalTwinCommandResponse.setPayload(objectMapper.writeValueAsString(object.body()));
+            digitalTwinCommandResponse.setPayload(new JsonPrimitive(objectMapper.writeValueAsString(object.body())));
```

Comment:
Can we just do 
```suggestion
            digitalTwinCommandResponse.setPayload(object);
```

here? Converting from jsonElement to string back to JsonPrimitive seems odd here

Reply:
How about changing to 
`digitalTwinCommandResponse.setPayload(new GsonBuilder().create().toJsonTree(object));`

`digitalTwinCommandResponse` takes JsonElement type payload only.

Reply:
And the catch block below for `JsonProcessingException` needs to remove as well.